Sources/TuistDependenciesTesting/Carthage/MockCarthageInteractor.swift

```diff
@@ -32,6 +32,32 @@ public final class MockCarthageInteractor: CarthageInteracting {
         }
     }
 
+    var invokedUpdate = false
+    var invokedUpdateCount = 0
+    var invokedUpdateParameters: UpdateParameters?
+    var invokedUpdateParametersList = [UpdateParameters]()
+    var stubbedUpdateError: Error?
```

Comment:
I know this is in multiple places and so this comment is not entirely specific to this PR but I honestly feel like these stub properties are usually waste of time and become quite unwieldy when you want to refactor the mock as you either have to do it by hand or incorporate some tool like Sourcery (which we should not ask of the contributors). I'd honestly stick to `updateStub: ((Input) -> Output)?` in most cases - unless you expect to re-use this often. But e.g. in this case `invokedUpdateCount` is used just once.